bundles/org.openhab.binding.fenecon/src/main/java/org/openhab/binding/fenecon/internal/FeneconHandler.java

```diff
@@ -0,0 +1,231 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fenecon.internal;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link FeneconHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Philipp Schneider - Initial contribution
+ */
+@NonNullByDefault
+public class FeneconHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FeneconHandler.class);
+
+    private FeneconConfiguration config = new FeneconConfiguration();
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private final List<String> channels = List.of("GridMode", "State", "EssSoc", "ConsumptionActivePower",
+            "ProductionActivePower", "GridActivePower", "EssDischargePower", "GridBuyActiveEnergy",
+            "GridSellActiveEnergy");
+
+    private final HttpClient httpClient;
+    private @Nullable Builder baseHttpRequest;
+
+    public FeneconHandler(Thing thing) {
+        super(thing);
+        httpClient = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(10)).build();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FeneconConfiguration.class);
+
+        logger.info("FENECON: initialize REST-API connection to {} with polling interval: {} sec", getBaseUrl(config),
+                config.refreshInterval);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        baseHttpRequest = createBaseHttpRequest(config);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 5, config.refreshInterval, TimeUnit.SECONDS);
```

Comment:
```suggestion
        pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 0, config.refreshInterval, TimeUnit.SECONDS);
```

Is tehre a need to wait 5 seconds?

Reply:
No, that was still from testing, I changed it.